fabfile/db/__init__.py

```diff
@@ -107,27 +130,29 @@ def postgres(fn):
         Decorator that adds PostgreSQL specific actions to task.
         """
         def wrapped(self, *args, **kwargs):
-            local(
-                'echo localhost:5432:postgres:%s:%s > ~/.pgpass'
-                % (self.db_cfg['USER'], self.db_cfg['PASSWORD'])
-            )
-            local(
-                'echo localhost:5432:%s:%s:%s >> ~/.pgpass'
-                % (self.db_cfg['NAME'], self.db_cfg['USER'], self.db_cfg['PASSWORD'])
-            )
-            local('chmod 600 ~/.pgpass')
+            self.host = self.host or (self.db_cfg['HOST'] if 'HOST' in self.db_cfg else 'localhost')
+            self.port = self.port or (self.db_cfg['PORT'] if 'PORT' in self.db_cfg else '5432')
+            with NamedTemporaryFile(suffix=".pgpass", delete=False) as f:
+                f.write('%s:%s:*:%s:%s' % (self.host,
+                                                  self.port,
+                                                  self.db_cfg['USER'],
+                                                  self.db_cfg['PASSWORD']))
+            os.environ["PGPASSFILE"] = f.name
             fn(self, *args, **kwargs)
-            local('rm ~/.pgpass')
+            local('rm %s ' % f.name)
         return wrapped
 
+
     @postgres
     def init_db_postgres(self, *args, **kwargs):
         """
         Init db to original state for postgres.
         """
         with lcd(self.base_path), settings(hide('warnings'), warn_only=True):
-            local('dropdb %s --username=%s -w ' % (self.db_cfg['NAME'], self.db_cfg['USER']))
-            local('createdb %s encoding="UTF8" --username=%s -w' % (self.db_cfg['NAME'], self.db_cfg['USER']))
+            local('dropdb %s --username=%s -w' % (self.db_cfg['NAME'],
```

Comment:
Alson we need to have user/pass args from console.
# VF: Also - we need arguments to pass user/password for restoration